tests/tests/test-staking/test-delegate-with-auto-compound.ts

```diff
@@ -172,6 +175,18 @@ describeDevMoonbeam("Staking - Delegate With Auto-Compound - wrong delegation hi
   });
 });
 
+describeDevMoonbeam("Staking - Delegate With Auto-Compound - 101%", (context) => {
+  it("should fail", async () => {
+    await expect(
+      context.createBlock(
+        context.polkadotApi.tx.parachainStaking
+          .delegateWithAutoCompound(alith.address, MIN_GLMR_DELEGATOR, 101, 0, 0, 0)
+          .signAsync(ethan)
+      )
+    ).to.eventually.be.rejectedWith("Value is greater than allowed maximum!");
```

Comment:
I'm confused about this, where is this error message supposed to come from?

I'm curious because we seem to accept anything over 100 and clamp it in all cases that I see.

Reply:
Ah, I see. Are you clear on when it's ok to use values over 100 and not? You have tests that explicitly cover cases where we can successfully use values over 100 and also cases where we can't, so I'm a bit confused about this.

What I'm worried about is us somehow letting a value over 100 find its way into our storage and then triggering this error during `Decode`.

Reply:
Actually I thought about it and it seems it's better to make the behavior consistent. Since the extrinsics are defined as `Percent` it will always fail on decode, but as we accept `u8` in our precompiles and `from_percent` silently caps the limits, we have this inconsistency. I updated the PR to also return an error in this case.
It's probably too late to fit into this runtime, but can be done in the next one since there's no disadvantage to us. Will be a breaking change though :warning: 

/cc @crystalin @librelois

Reply:
Nothing will. `from_percent` guarantees to cap values in `[0,100]`, so we're safe.

Reply:
You're right, I don't think that's possible.